semanticdb/metai/src/main/scala/scala/meta/internal/metai/Main.scala

```diff
@@ -0,0 +1,100 @@
+package scala.meta.internal.metai
+
+import java.net.URI
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.nio.file.StandardOpenOption
+import scala.collection.mutable
+import scala.meta.cli.Reporter
+import scala.meta.internal.io.FileIO
+import scala.meta.internal.io.PathIO
+import scala.meta.internal.io.PlatformFileIO
+import scala.meta.internal.{semanticidx => i}
+import scala.meta.internal.{semanticdb => s}
+import scala.meta.internal.semanticdb.Scala._
+import scala.meta.internal.io._
+import scala.meta.io.AbsolutePath
+import scala.meta.io.Classpath
+import scala.meta.io.RelativePath
+import scala.meta.metai.Settings
+import scala.util.control.NonFatal
+
+final class Main(settings: Settings, reporter: Reporter) {
+  def process(): Option[Classpath] = {
```

Comment:
If Metai modifies the classpath in place, I suggest that we don't pretend that it's `Classpath => Classpath` and just return a boolean for success or failure. If you agree, I suggest we also don't print anything to the console upon successful completion.